src/main/resources/com/google/api/codegen/csharp/gapic_resourcenames.snip

```diff
@@ -292,13 +292,13 @@
         public override string ToString() => Name.ToString();
 
         /// <inheritdoc />
-        public override int GetHashCode() => ToString().GetHashCode();
+        public override int GetHashCode() => Name.GetHashCode();
 
         /// <inheritdoc />
         public override bool Equals(object obj) => Equals(obj as {@resource.typeName});
 
         /// <inheritdoc />
-        public bool Equals({@resource.typeName} other) => ToString() == other?.ToString();
+        public bool Equals({@resource.typeName} other) => Name.Equals(other?.Name);
```

Comment:
Following on from offline discussion, the hashcode and equality changes have been reverted.